pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -57,36 +57,31 @@ def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg,
         cc_none = subclass_where(BaseDualPseudoController,
                                  pseudo_controller_name='none')
 
-        # Multigrid types and suitable config files for lower multigrid levels
-        multip_types = {}
-        mgcfgs = {l: Inifile(cfg.tostr()) for l in self.levels}
-        for l, mgcfg in mgcfgs.items():
+        # Suitable pseudo-integrators for all levels
+        self.pintgs = {}
+        for l in self.levels:
+            mgcfg = Inifile(cfg.tostr())
             mgcfg.set('solver', 'order', l)
             mgcfg.set(sect, 'pseudo-dt', self.dtaus[l])
-            name = '_'.join(['multiPpseudointegrator' + str(l)])
 
             if l == self._order:
-                bases = [(cn, cc), (pn, pc)]
+                bases = (cc, pc)
 
-                multip_types[l] = type(name, tuple(bc for bn, bc in bases),
-                                       dict(name=name))
+                # Get the convergence monitoring method
+                self.mg_conv_mon = cc.conv_mon
             else:
-                bases = [('none', cc_none), (pn, pc)]
+                bases = (cc_none, pc)
 
                 for sec in cfg.sections():
                     m = re.match(r'solver-(.*)-mg-p{0}'.format(l), sec)
                     if m:
                         mgcfg.rename_section(m.group(0), 'solver-' + m.group(1))
 
-            multip_types[l] = type(name, tuple(bc for bn, bc in bases),
-                                   dict(name=name))
 
-        # Get pseudo-integrators at all multigrid levels
-        self.pintgs = {}
-        for i, psint in multip_types.items():
             # A class that bypasses pseudo-controller methods within a cycle
-            class lpsint(psint):
-                aux_nregs = 2 if i != self._order else 0
+            class lpsint(*bases):
+                name = '_'.join(['multiPpseudointegrator' + str(l)])
```

Comment:
The join is not required here as the list only has a single element.

name = 'MultiPPseudoIntegrator' + str(l)

will work.